mitodl/mitxpro
# feat: welcome emails for xPRO Learners

### What are the relevant tickets?
[#2470](https://github.com/mitodl/hq/issues/2470)

### Description (What does it do?)
This PR implements the welcome email feature, which sends a welcome email based on each user's course enrollment.

### How can this be tested?

1. Checkout to this branch and make sure openedx is configured.
2. Try to enroll any user to any course through frontend, or you can use create enrollment command to enroll a user.
   - docker-compose exec web ./manage.py create_enrollment --user "<email>" --run "<course_run>" --code="test7e91c44ea5a999beasdfadsfadfadfwer"

3. Check if you receive welcome email along with enrollment email.
4. Verify the content of the email with welcome letter provided in the the ticket https://github.com/mitodl/hq/issues/2470#issue-1912198810 or you can see the link here: https://docs.google.com/document/d/1gGdXSVT4U34_bIADqGzZvcEyWuulM6VK8VDbqyDF1MY/edit
5.  Now, again try to enroll user with transfer enrollment, and follow steps 3 and 4.

### Screenshots (if appropriate):
- See screenshot here:
![image](https://github.com/mitodl/mitxpro/assets/41048311/8b8e0a9b-b7c2-4f81-93fd-9d75b4dfc7d0)


- To see the downloaded email as PDF:
[Arbisoft Mail - Welcome! course-v1_edX+DemoX+Demo_Course _ Demonstration Course starts on Feb 5, 2013.pdf](https://github.com/user-attachments/files/16053623/Arbisoft.Mail.-.Welcome.course-v1_edX%2BDemoX%2BDemo_Course._.Demonstration.Course.starts.on.Feb.5.2013.pdf)

Note: The logo is missing from the email attachment above because the links point to a local server. However,
you can refer to the screenshot below to review the logo, which should ideally be the same once deployed:

<img width="724" alt="image" src="https://github.com/mitodl/mitxpro/assets/41048311/1247abdc-c610-47de-9c39-de9c6948b561">


### Additional Context
